staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/multilock.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcelock
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// MultiLock is used for lock's migration
+type MultiLock struct {
+	Primary   Interface
+	Secondary Interface
+}
+
+// Get returns the older election record of the lock
+func (ml *MultiLock) Get() (*LeaderElectionRecord, error) {
+	primary, err := ml.Primary.Get()
+	if err != nil {
+		return nil, err
+	}
+	secondary, err := ml.Secondary.Get()
+	if err != nil {
+		return nil, err
+	}
+	// check if the older resource lock is expired
+	if secondary.RenewTime.Before(&primary.RenewTime) {
+		return secondary, nil
+	}
+	return primary, nil
+}
+
+// Create attempts to create both primary lock and secondary lock
+func (ml *MultiLock) Create(ler LeaderElectionRecord) error {
+	err := ml.Primary.Create(ler)
+	if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
If we assume, that there never should be situation that Primary doesn't exist and Secondary exists (which I think is actually a valid asusmption), then masking Primary sounds fine to me.
But this has to be very clearly documented at the very least.

@liggitt - WDYT?

Reply:
> If we assume, that there never should be situation that Primary doesn't exist and Secondary exists

Won't that be the case if another of the members is an n+1 controller that only pays attention to Lease objects?

Reply:
My assumption for that case was that there are two cases:
- either at some point there was "n" controller, which already created those objects (so basically we are upgrading) [technically it can also be removed by sth, but I wouldn't consider that as a high risk]
- or there weren't (and we create directly cluster in n+1 version) and that's not a problem

Reply:
That doesn't seem like a good assumption. two-object leader election needs to be bulletproof (within the assumptions that the electing clients are well-behaved and aren't deleting each other's objects)